src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.cc

```diff
@@ -0,0 +1,687 @@
+}
+
+std::ostream &OMapInnerNode::print_detail_l(std::ostream &out) const
+{
+  return out << ", size=" << get_size()
+	     << ", depth=" << get_meta().depth;
+}
+
+OMapInnerNode::handle_splitted_ret
+OMapInnerNode::handle_splitted(omap_context_t oc, internal_iterator_t iter,
+                               mutation_result_t mresult)
+{
+  logger().debug("{}: {}","OMapInnerNode",  __func__);
+  auto [left, right, pivot] = *(mresult.split_tuple);
+  if (!extent_will_overflow(pivot.size() + 1, std::nullopt)) {
+    if (!is_pending()) {
+      auto mut = oc.tm.get_mutable_extent(oc.t, this)->cast<OMapInnerNode>();
+      auto mut_iter = mut->iter_idx(iter.get_index());
+      return mut->handle_splitted(oc, mut_iter, mresult);
+    }
+    journal_inner_update(iter, left->get_laddr(), maybe_get_delta_buffer());
+    journal_inner_insert(iter + 1, right->get_laddr(), pivot,
+                         maybe_get_delta_buffer());
+    return insert_ret(
+           insert_ertr::ready_future_marker{},
+           mutation_result_t(mutation_status_t::SUCCESS, std::nullopt, std::nullopt));
+  } else {
+    return make_split_children(oc)
+      .safe_then([this, oc, iter, mresult = std::move(mresult)] (auto tuple) {
+       OMapNodeRef pleft = std::get<0>(tuple);
+       OMapNodeRef pright = std::get<1>(tuple);
+       std::string p_pivot = std::get<2>(tuple);
+       OMapInnerNodeRef p_left = pleft->cast<OMapInnerNode>();
+       OMapInnerNodeRef p_right = pright->cast<OMapInnerNode>();
+       auto [c_left, c_right, c_pivot] = *(mresult.split_tuple);
+       if (c_pivot < p_pivot) {  //left
+         auto mut_iter = p_left->iter_idx(iter.get_index());
+         p_left->journal_inner_update(mut_iter, c_left->get_laddr(),
+                                      p_left->maybe_get_delta_buffer());
+         p_left->journal_inner_insert(mut_iter + 1, c_right->get_laddr(), c_pivot,
+                                      p_left->maybe_get_delta_buffer());
+       } else if(c_pivot == p_pivot) {
+         auto mut_iter = p_left->iter_idx(iter.get_index());
+         p_left->journal_inner_update(mut_iter, c_left->get_laddr(),
+                                      p_left->maybe_get_delta_buffer());
+         mut_iter = p_right->iter_idx(0);
+         p_right->journal_inner_update(mut_iter, c_right->get_laddr(),
+                                       p_right->maybe_get_delta_buffer());
+       } else {  //right
+         auto mut_iter = p_right->iter_idx(iter.get_index() - p_left->get_node_size());
+         p_right->journal_inner_update(mut_iter, c_left->get_laddr(),
+                                       p_right->maybe_get_delta_buffer());
+         p_right->journal_inner_insert(mut_iter + 1, c_right->get_laddr(), c_pivot,
+                                         p_right->maybe_get_delta_buffer());
+       }
+       if (get_node_meta().depth == oc.omap_root.depth) {
+         return root_splitted(oc, get_laddr(), get_node_meta().depth, tuple);
```

Comment:
I think root_splitted (which should be renamed to handle_root_split) and the corresponding merge handler should be moved to btree_omap_manager.